test/unit/assorted/optional_require.test.ts

```diff
@@ -0,0 +1,72 @@
+import { expect } from 'chai';
+import { existsSync } from 'fs';
+import { resolve } from 'path';
+
+import {
+  AuthContext,
+  compress,
+  GSSAPI,
+  HostAddress,
+  MongoDBAWS,
+  MongoMissingDependencyError
+} from '../../mongodb';
+
+function moduleExistsSync(moduleName) {
+  return existsSync(resolve(__dirname, `../../../node_modules/${moduleName}`));
+}
+
+describe('optionalRequire', function () {
+  describe('Snappy', function () {
+    it('should error if not installed', async function () {
+      const moduleName = 'snappy';
+      if (moduleExistsSync(moduleName)) {
+        return this.skip();
+      }
+
+      const error = await compress(
+        { zlibCompressionLevel: 0, agreedCompressor: 'snappy' },
+        Buffer.alloc(1)
+      ).then(
+        () => null,
+        e => e
+      );
+
+      expect(error).to.be.instanceOf(MongoMissingDependencyError);
+    });
+  });
+
+  describe('Kerberos', function () {
+    it('should error if not installed', async function () {
+      const moduleName = 'kerberos';
+      if (moduleExistsSync(moduleName)) {
+        return this.skip();
+      }
+      const gssapi = new GSSAPI();
+
+      const error = await gssapi
+        .auth(new AuthContext(null, true, { hostAddress: new HostAddress('a'), credentials: true }))
+        .then(
+          () => null,
+          e => e
+        );
+
+      expect(error).to.be.instanceOf(MongoMissingDependencyError);
+    });
+  });
+
+  describe('aws4', function () {
+    it('should error if not installed', async function () {
+      const moduleName = 'aws4';
```

Comment:
Now that AWS 4 is a dev dependency, this test never runs (right?).  Same with snappy.  Kerberos should be running because it isn't tracked as a dev dependency.

Do we need to add a variant in CI that tests without these packages installed to confirm we have coverage here?